### ISSUE TITLE
Make sure all tests are working

### ISSUE THREAD

**Issue by bikegeek** [dtcenter/METplotpy#73]
Verify and fix if needed, the tests that are in the METplotpy/test directory in anticipation of automation.

### Time Estimate ###
~1-2 days if fixes/updates are required

### Sub-Issues ###
Tests that are in the METplotpy/test directory:
- [x] Line plot
- [x] Reliability plot*   _see comments below for reliability plot comments_
- [x] Performance Diagram
- [x] Histogram *Made updates to test
- [x] Difficulty Index* _see comments below for issues with this test_
- [x] Equivalence Testing Bounds

### Funding Source ###
2702691

### Assignee ###
- [x] Select **engineer(s)** Minna et.al
- [x] Select **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [X] Add/update log messages for easier debugging. N/A
- [x] Add/update unit tests.
- [x] Add/update documentation.N/A
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Close this issue.

**Comment by bikegeek** [comment]
Line Plot Tests OK
-------------------
Verified that the tests for line plot under METplotpy/test/line are working via command line by setting PYTHONPATH to $YOURHOME/METplotpy:$YOURHOME/METplotpy/metplotpy and via pycharm ide by setting 'Sources' to METplotpy, METplotpy/metplotpy, METplotpy/metplotpy/plots

**Comment by bikegeek** [comment]
Reliability Plot Tests Pass 
--------------------------

- Successfully run from command line with PYTHONPATH set to:

$YOURHOME/METplotpy:$YOURHOME/METplotpy/metplotpy 

- Suppress FutureWarnings, which don't impact us:
- import warning
- warnings.simplefilter(action='ignore', category=FutureWarning)

**Comment by bikegeek** [comment]
Performance Diagram Tests Pass
---------------------------------
**After modifying the imports to performance_diagram.py from:**

from performance_diagram_config import PerformanceDiagramConfig
from performance_diagram_series import PerformanceDiagramSeries

**TO**

from . performance_diagram_config import PerformanceDiagramConfig
from . performance_diagram_series import PerformanceDiagramSeries

to explicitly indicate that the modules are in parallel folder structure of METplotpy/metplotpy/plots

**Also changed import in test_performance_diagram.py** by adding:

sys.path.append('../..')

Running successfully via Pycharm IDE and from command line.

Running by command line:
--------------------------
PYTHONPATH set to:
$YOUR_DIR/METplotpy:$YOUR_DIR/METplotpy/metplotpy

**Comment by bikegeek** [comment]
### **Histogram**

- to ensure this test consistently works, explicitly set the METPLOTPY_BASE env via os.environ['METPLOTPY_BASE'] = "../../metplotpy"

**Comment by bikegeek** [comment]
### **Difficulty Index**

- "accepted" plot, provided by NRL is no longer a 100% match with the plot generated by the test
- The title font appearance is the only difference from the "accepted" plot
- The comparison.mssim value is 0.9926... rather than 1 (value of 1 reflects a perfect match)
- sent email to Lindsay to determine how to proceed:
- contact NRL and see if this change in title font is acceptable, and then modify assert to check for comparison.mssim value greater than 0.99 (more than 99% match) OR
- replace "accepted" plot with the newly created plot

**Solution**
- run test in conda environment that has matplotlib 3.3.3, the conda env that I am using is 3.3.1
- add this version of matplotlib as a requirement for METplotpy

-

**Comment by bikegeek** [comment]
### **Grid to Grid**

- changed the import statement to: import contributed.grid_to_grid.plot_grid_to_grid as pgg (in feature_73_METplotpy_tests feature branch
- test_expected_filesizes() passes
- test_expected_files_created() fails

**Comment by bikegeek** [comment]
### **Series Analysis**

**Errors:**
- ModuleNotFoundError: plot_series_by_init
- ModuleNotFoundError: plot_series_by_lead_all

**Comment by bikegeek** [comment]
Difficulty index test modified and successfully runs.

**Comment by bikegeek** [comment]
### **Grid to Grid**
- **test_expected_files_created() fails**
- changed the import statement to: _import contributed.grid_to_grid.plot_grid_to_grid as pgg_ (in feature_73_METplotpy_tests feature branch
- **test_expected_filesizes() passes**


### **Series Analysis**
**Tests that Pass:**

test_plot_series_by_init
**Tests that Fail:**

test_plot_series_by_lead_all.py has assertion errors. It also looks like tests are using input and output that reside in /d1/METplus_Plotting_Data. The data needs to be moved to 'kiowa' under /d1/projects/METplotpy/METplotpy_Data/series_analysis and/or made available 'locally' in the source code (only if data files are very small).
Modifications:

ModuleNotFoundError: plot_series_by_init -> modified import to use sys.path.append to implicitly point to the METplotpy/metplotpy/contributed folder

ModuleNotFoundError: plot_series_by_lead_all -> modified import to use sys.path.append to implicitly point to the METplotpy/metplotpy/contributed folder

### Update 3/19/2021:
data that was on **kiowa** _/d1/projects/METplus/METplus_Plotting_Data/series_by_init_ (and series_by_lead, series_by_grouping) is gone. Re-create data that can be used for running the test and save this in the /d1/projects/METplus/METplus_Plotting_Data directory which will someday be part of the release tarball.

### Update 4/2/2021
extensive cleaning up of the Grid-to-Grid and series analysis plotters (they were not tested in this repository and the tests had to be refactored and the plotting scripts were refactored to use a simple config file for easier set-up in the future).  Tests working on Mac OS.
